### PR TITLE
Add Basic User Interface

### PR DESCRIPTION
**Pull request by Epaltechs**
### In this part of the project, I make commits with the following implementations:
- Create an app.rb file that serves as the console app entry-point with methods that do the following:
   - List all books.
   - List all people.
   - Create a person (teacher or student, not a plain Person).
   - Create a book.
   - Create a rental.
   - List all rentals for a given person's id.
- In main.rb I define the entry point with a method called main that does the following:
   - Present the user with a list of options to perform.
   - Lets users choose an option.
   - If needed, ask for parameters for the option.
   - Create an exit to quit the app.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  puts 'Do you want create student (1) or teacher (2)? [Enter the number]: '
+  person_type = gets.chomp.to_i
+  case person_type
+  when 1
+    print 'Enter student name: '
+    name = gets.chomp
+    print 'Enter age: '
+    age = gets.chomp
+    @people.push(Student.new('classroom', age, name))
+    puts 'Student created successfully'
```
Comment (roshan-bajgain) [inline app.rb]: - [ ] While creating the age of a student there is an error `push` is not defined kindly fix it.
### Error
sssss

**Review by roshan-bajgain — CHANGES_REQUESTED** [review]
Hi @Epaltechs, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️
- Professional README file. ✔️
- PR has a good title and summary. ✔️

## Required Changes ♻️

- [ ] I cannot leave comments there so I am leaving them here. While Entering specialization on the teacher I got the error. See this error below and kindly fix it
### Error
ssssss

- [ ] when I try to create rentals it is showing an error as you can see in the rental.rb file. undefined method `rentals'. Please fix the error
- [ ] The rental is not listening kindly solve the problem as well.
### Error

ss

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Epaltechs** [comment]
### In this commit, I implement the changes requested by the code review.

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  rentals.push(Rental.new(provided_date, people[selected_person], books[selected_book]))
+  puts 'Rental created successfully.'
+end
+
+def list_all_rentals_by_id(rentals)
+  print 'ID of person: '
+  person_id = gets.chomp.to_i
+  rentals.map do |rental|
+    puts "Date: #{rental.date}, Book: #{rental.book.title} by #{rental.book.author}" if rental.person.id == person_id
+  end
```
Comment (gemmen29) [inline app.rb:77]: - [ ] Kindly fix this error in the `List all rentals for a given person id` option as requested by the previous reviewer 
image

File: main.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+def main_menu
+  puts 'Welcome to School library App!'
+  puts 'Please choose an option by entering a number:
+  1 - List all books
+  2 - List all people
+  3 - Create a person
+  4 - Create a book
+  5 - Create a rental
+  6 - List all rentals for a given person id
+  7 - Exit'
```
Comment (gemmen29) [inline main.rb:20]: - [ ] [OPTIONAL] Kindly make the `Exit` work as now the program continues when I press this case.
image

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi @Epaltechs, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## To Highlight 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Epaltechs** [comment]
### In this commit, I have fixed the error in the code as requested by the code review.

**Review by zieeco — APPROVED** [review]
# status: APPROVED :green_circle: :100: 

Hi, @Epaltechs 


68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f78543078657a514755357843444a754350652f67697068792e676966


Good job so far 👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**